Implement a mechanism for registering and marking functions as thread-unsafe (#37)
Quansight-Labs/pytest-run-parallel: README.rst

        ...
    ]

> Similarly, if a set of functions call are known to be thread unsafe and should
> cause a test to be marked as well, their fully-qualified names can be
> registered through the `thread_unsafe_functions` option in the INI file
> (or under `tool.pytest.ini_options` when using `pyproject.toml`):

Reviewer: ```suggestion
Similarly, if a function is known to be thread unsafe and should
cause a test to be marked as thread-unsafe as well, the fully-qualified names 
of thread-unsafe functions can be registered through the 
`thread_unsafe_functions` option in the INI file (or under 
`tool.pytest.ini_options` when using `pyproject.toml`):
```

.. code-block:: ini

    [pytest]
    thread_unsafe_functions =
        module.submodule.func1
        module.submodule2.func2
        ...

> Also, functions that declare the assignment `__thread_unsafe__ = True` that get
> called by a test and are up to two levels below in the call stack will
> automatically cause such test to be marked as thread unsafe.

Reviewer: ```suggestion
Also, if you define a `__thread_unsafe__` attribute on a function that is
called by a test and is up to two levels below in the call stack, then 
pytest-run-parallel will automatically detect that a thread-unsafe function
is being used and will mark the test as thread-unsafe.
```

I think this is what you were going for? It might also be worth mentioning that the plugin does this with AST parsing and isn't a runtime thing.

Also separately can you just make it so that they need to define a `__thread_unsafe__` attribute and it can have any value? Otherwise it probably makes more sense to spell it `__thread_safe__ = False`, since people have an easier time parsing boolean flags that default to True.
